bestbook/views/__init__.py

```diff
@@ -120,11 +120,28 @@ class (e.g. Ask, Login, Observe, Submit)
             "submit": Submit,
         }
         form = request.form
+
         return jsonify(forms[resource]().post())
 
 # API POST Handlers
 
+def require_auth(func):
+    def authed_func(*args, **kwargs):
+        # username needs to be added to the front-end forms
+        # as a hidden field (populated by session.get('username')
+        # Note: we may not want to keep username + s3 in request.form
+        username = request.form.get('username')
```

Comment:
I think that this will fail for external POSTs.  Observations are coming from OL as JSON (`request.data` instead of `request.form`).  I agree with your point about getting the username from the authenticate function.  It looks like the `auth_config` object will have the [username](https://github.com/jjjake/internetarchive/blob/feba1086ee609f8b24acfe24b548c2390b9067b5/internetarchive/config.py#L68-#L70).

Working on sending the S3 keys in OL POST headers this evening.  I'll take a closer look at this afterwards.

Reply:
+1